Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -172,26 +190,24 @@ private func parseBytes(
         var padding = 0 // # padding '=' chars
         digits: while true {
             let digit = source[index]
-            let k = base64Values[Int(digit)]
-            if k >= 0 {
-                n <<= 6
-                n |= k
-                chars += 1
-                if chars == 4 {
-                    p[0] = UInt8(truncatingIfNeeded: n >> 16)
-                    p[1] = UInt8(truncatingIfNeeded: n >> 8)
-                    p[2] = UInt8(truncatingIfNeeded: n)
-                    p += 3
-                    chars = 0
-                    n = 0
-                }
-            } else {
+            var k = base64Values[Int(digit)]
+            if k < 0 {
                 switch digit {
                 case asciiDoubleQuote:
                     source.formIndex(after: &index)
                     break digits
+                case asciiBackslash:
```

Comment:
While we are here, I notice that the protobuf documentation permits the base64 string to be URL-safe, i.e. `+` and `=` and `/` can by `%2B` or whatever. Maybe open another issue?

Reply:
They are both mixed in the table, the section 4/5 checks are to ensure we only ever get one set and not both (if I'm remembering correctly)

Reply:
"URL-safe base-64" is not the same as "URL encoding."  The URL-safe alphabet is the one that uses "-" and "_" instead of "+" and "/".  (This is formalized in RFC4648 section 5 as the "Base 64 Encoding with URL and Filename Safe Alphabet" aka "base64url".)

I've expanded the comments above the table to explain this further.

Reply:
Ah, my mistake. Thanks.